pytests/tests/test_checksum.py

```diff
@@ -0,0 +1,117 @@
+# Copyright (C) 2019 - 2022 VMware, Inc. All Rights Reserved.
+#
+# Licensed under the GNU General Public License v2 (the "License");
+# you may not use this file except in compliance with the License. The terms
+# of the License are located in the COPYING file of this distribution.
+#
+
+import os
+import shutil
+import re
+import filecmp
+import pytest
+
+WORKDIR = '/root/repofrompath/workdir'
+
+
+@pytest.fixture(scope='module', autouse=True)
+def setup_test(utils):
+    yield
+    teardown_test(utils)
+
+
+def teardown_test(utils):
+    utils.run(['tdnf', 'erase', '-y', 'tdnf-test-one'])
+    if os.path.isdir(WORKDIR):
+        shutil.rmtree(WORKDIR)
+
+
+def enable_and_create_repo(utils):
+    workdir = WORKDIR
+    utils.makedirs(workdir)
+    reponame = 'photon-test-sha512'
+
+    ret = utils.run(['tdnf', '-v', '--disablerepo=*', '--enablerepo=photon-test-sha512', 'makecache'])
+    assert ret['retval'] == 0
+
+    ret = utils.run(['tdnf', '--repo={}'.format(reponame),
+                     '--download-metadata',
+                     'reposync'],
+                    cwd=workdir)
+    assert ret['retval'] == 0
+    synced_dir = os.path.join(workdir, reponame)
+    assert os.path.isdir(synced_dir)
+    assert os.path.isdir(os.path.join(synced_dir, 'repodata'))
+    assert os.path.isfile(os.path.join(synced_dir, 'repodata', 'repomd.xml'))
+
+
+def find_suffix(filename, prefix):
+    escaped_prefix = re.escape(prefix)
+
+    pattern = rf'^{escaped_prefix}(.+)$'
+
+    match = re.search(pattern, filename)
+    if match:
+        return match.group(1)
+    else:
+        return None
+
+
+def modify_rpm(workdir, orig_pkg, new_pkg):
+    orig_filename = ''
+    orig_path = ''
+
+    for filename in os.listdir(workdir):
+        if filename.startswith(orig_pkg):
```

Comment:
If we find `new_pkg` before `orig_pkg`, this won't work because it wont't set `orig_path`, right?

This function is very confusing. Please rewrite it with a simpler version.